erpnext/accounts/doctype/journal_entry/journal_entry.py

```diff
@@ -98,6 +100,16 @@ def update_advance_paid(self):
 			for voucher_no in list(set(order_list)):
 				frappe.get_doc(voucher_type, voucher_no).set_total_advance_paid()
 
+	def update_status_for_full_and_final_statement(self):
+		fnf_docname = [d.reference_name for d in self.accounts if d.reference_type=="Full And Final Statement"]
+		if len(fnf_docname):
+			fnf_docname = fnf_docname[0]
+		if self.docstatus == 1:
+			frappe.db.set_value("Full And Final Statement", fnf_docname, "status", "Paid")
+		if self.docstatus == 2:
+			frappe.db.set_value("Full And Final Statement", fnf_docname, "status", "Unpaid")
```

Comment:
You don't need to make a list and then choose the first element. You can directly iterate and update

```suggestion
		for entry in self.accounts:
			if entry.reference_type == "Full And Final Statement":
				if self.docstatus == 1:
					frappe.db.set_value("Full And Final Statement", entry.reference_name, "status", "Paid")
				elif self.docstatus == 2:
					frappe.db.set_value("Full And Final Statement", entry.reference_name, "status", "Unpaid")
```

Reply:
done